### PR TITLE
Firefox: Pass browser context cookie tests

### LINKED ISSUE #94: Pass cookies.spec.js tests in Firefox

### PR DESCRIPTION
**Pull request by Meir017**
closes #94

### CODE REVIEW

**Comment by kblok** [comment]
closes #94 @Meir017 ?

File: src/PlaywrightSharp.Firefox/FirefoxBrowserContext.cs [hunk 1]
```diff
@@ -56,13 +64,33 @@ public async Task CloseAsync()
...
+            {
+                BrowserContextId = _browserContextId,
+                Cookies = Array.ConvertAll(cookies, c => (CookieOptions)c),
+            });
+
+        /// <inheritdoc cref="IBrowserContextDelegate.GetCookiesAsync"/>
+        public async Task<NetworkCookie[]> GetCookiesAsync()
+        {
+            var result = await _connection.SendAsync(new BrowserGetCookiesRequest { BrowserContextId = _browserContextId }).ConfigureAwait(false);
+            return Array.ConvertAll(result.Cookies, c => (NetworkCookie)c);
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/FirefoxBrowserContext.cs:90]:
```suggestion
            return result.Cookies.Cast<NetworkCookie>();
```

And we could return an `IEnumerable` so we don't cast all of them if we don't need to.

File: src/PlaywrightSharp.Firefox/Protocol/CookieOptions.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            Url = cookie.Url,
+            Domain = cookie.Domain,
+            Path = cookie.Path,
+            Secure = cookie.Secure,
+            HttpOnly = cookie.HttpOnly,
+            SameSite = ConvertSameSite(cookie.SameSite),
+            Expires = cookie.Expires,
+        };
+
+        internal static CookieOptionsSameSite? ConvertSameSite(SameSite? sameSite)
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/Protocol/CookieOptions.cs:31]: 'ConvertToCookieOptionsSameSite'

File: src/PlaywrightSharp.Firefox/Protocol/CookieOptions.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            => sameSite switch
+            {
+                PlaywrightSharp.SameSite.None => CookieOptionsSameSite.None,
+                PlaywrightSharp.SameSite.Lax => CookieOptionsSameSite.Lax,
+                PlaywrightSharp.SameSite.Strict => CookieOptionsSameSite.Strict,
+                _ => null,
+            };
+
+        internal static SameSite? ConvertSameSite(CookieOptionsSameSite? sameSite)
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/Protocol/CookieOptions.cs:40]: 'ConvertToSameSite'

**Comment by kblok** [comment]
Linq task won't work? 🤔

**Comment by Meir017** [comment]
I don't think so https://stackoverflow.com/a/13316745/7025656

**Pull request merged**